CRM/Core/Payment.php

```diff
@@ -684,20 +684,25 @@ public static function handlePaymentMethod($method, $params = array()) {
     $method = 'handle' . $method;
     $extension_instance_found = FALSE;
 
-    // In all likelihood, we'll just end up with the one instance returned here. But it's
-    // possible we may get more. Hence, iterate through all instances ..
+    // In all likelihood, we'll just end up with the one instance
+    // returned here. But it's possible we may get more. Hence,
+    // iterate through all instances ..
 
     while ($dao->fetch()) {
-      // Check pp is extension - is this still required - surely the singleton below handles it.
+      // Check pp is extension - is this still required - surely the
+      // singleton below handles it.
       $ext = CRM_Extension_System::singleton()->getMapper();
       if ($ext->isExtensionKey($dao->class_name)) {
         $paymentClass = $ext->keyToClass($dao->class_name, 'payment');
         require_once $ext->classToPath($paymentClass);
       }
+      else {
+        $paymentClass = $dao->class_name;
```

Comment:
Two things with this line:
 1. The `class_name` that's stored in the DB is not a full class name. (You can use `select class_name from civicrm_payment_processor_type` to see what I mean.)
 2. The `$paymentClass` doesn't seem to be used anywhere else. If you search for `paymentClass` in this file, it only shows up in the context of handling legacy `payment` extensions.